animations/lib/src/misc/card_swipe.dart

```diff
@@ -0,0 +1,158 @@
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/physics.dart';
+
+class CardSwipeDemo extends StatefulWidget {
+  static String routeName = '/misc/card_swipe';
+
+  @override
+  _CardSwipeDemoState createState() => _CardSwipeDemoState();
+}
+
+class _CardSwipeDemoState extends State<CardSwipeDemo> {
+  var fileNames = [
+    'assets/eat_cape_town_sm.jpg',
+    'assets/eat_new_orleans_sm.jpg',
+    'assets/eat_sydney_sm.jpg',
+  ];
+
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Card Swipe'),
+      ),
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: <Widget>[
+            Container(
+              height: 500,
+              child: Stack(
+                children: <Widget>[
+                  for (final fileName in fileNames)
+                    SwipeableCard(
+                      image: fileName,
+                      onSwiped: () {
+                        setState(() {
+                          fileNames.remove(fileName);
+                        });
+                      },
+                    ),
+                ],
+              ),
+            ),
+            RaisedButton(
+              child: const Text('Refill'),
+              onPressed: () {
+                setState(() {
+                  fileNames = [
+                    'assets/eat_sydney_sm.jpg',
+                    'assets/eat_cape_town_sm.jpg',
+                    'assets/eat_new_orleans_sm.jpg',
+                  ];
+                });
+              },
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class Card extends StatelessWidget {
+  final String image;
+
+  Card(this.image);
+
+  Widget build(BuildContext context) {
+    return AspectRatio(
+      aspectRatio: 3 / 5,
+      child: DecoratedBox(
+        decoration: BoxDecoration(
+          borderRadius: BorderRadius.circular(20.0),
+          image: DecorationImage(
+            image: AssetImage(image),
+            fit: BoxFit.cover,
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class SwipeableCard extends StatefulWidget {
+  final String image;
+  final VoidCallback onSwiped;
+
+  SwipeableCard({
+    this.onSwiped,
+    this.image,
+  });
+
+  _SwipeableCardState createState() => _SwipeableCardState(this.image);
+}
+
+class _SwipeableCardState extends State<SwipeableCard>
+    with SingleTickerProviderStateMixin {
+  AnimationController _controller;
+  Animation<Offset> _animation;
+  double _dragStartLocalPosition, _currentPosition;
+  final String image;
```

Comment:
Did this comment get lost in the shuffle? Keeping data that doesn't need to mutate in the Widget is definitely the best practice, so that's where the field needs to be unless I'm missing something.

Reply:
yeah I missed this one. fixing...